lib/active_merchant/billing/gateways/payflow.rb

```diff
@@ -188,6 +189,53 @@ def build_credit_card_request(action, money, credit_card, options)
           end
         end
         xml.target!
+        add_level_two_three_fields(xml, options)
+      end
+
+      def add_level_two_three_fields(xml, options)
+        if options[:level_two_fields] || options[:level_three_fields]
+          xml_doc = Nokogiri::XML.parse(xml.target!) ## so it doesn't put on an xml instruction
+          %i[level_two_fields level_three_fields].each do |fields|
+            xml = add_fields(xml_doc, options[fields]) if options[fields]
```

Comment:
If both l2 and l3 field options are passed, won't this effectively discard the former after we finish processing the latter? Is that intentional? (Like, are they mutually exclusive?) Or does the call to `add_fields` mutate the passed `xml_doc` such that the second time through the resultant string is built up in addition to the first pass (so that the final result includes the processing from both passes)? (I am suspecting the latter, but just want to make sure)

Reply:
Nevermind on this. It's clear to me upon further reading that we are updating the `xml_doc` with each pass, so the return value is always the built-up value regardless of whether we call `add_fields` once or twice (depending on the passed options). I see your tests also verify this behavior. 👍